### PR TITLE
Improve static dependency and using tutorials

### LINKED ISSUE #24: Static configuration capability enhancement of Servicefence.Spec
Lazyload supports dynamic updating of service dependencies via metric, but in some scenarios, users want to add some static service dependencies when lazy loading is enabled, such as all services under a namespace, or services with certain labels. 
We will carefully consider the actual requirements and implement similar functionality through fence's spec field.

### PR DESCRIPTION
**Pull request by cywang1905**
fix #24 

- support config ns level in spec.host
- add doc

```yaml 
Spec Example
   spec:
    enable: true
    host:
      reviews.default.svc.cluster.local: # static dependency of reviews.default service
        stable:
      test/*: {} # static dependency of services in test namespace
    labelSelector: # Match service label, multiple selectors are 'or' relationship, static dependency
      - selector:
          project: back
      - selector: # labels in same selector are 'and' relationship
          project: front
          group: web

### CODE REVIEW

File: controllers/servicefence_controller.go [hunk 1]
```diff
@@ -376,17 +375,61 @@ func addDomainsWithHost(domains map[string]*lazyloadv1alpha1.Destinations, sf *l
 
 	for h, strategy := range sf.Spec.Host {
 
+		// handle namespace level host, like 'default/*'
```
Comment (YonkaFang) [inline controllers/servicefence_controller.go:378]: split into a separate function like `fulfillByNs`...

File: controllers/servicefence_controller.go [hunk 2]
```diff
@@ -376,17 +375,61 @@ func addDomainsWithHost(domains map[string]*lazyloadv1alpha1.Destinations, sf *l
...
+			if len(subHosts) != 2 {
+				log.Errorf("%s is invalid host, skip", h)
+				continue
+			}
+			nsSvcCache.RLock()
+			svcs := nsSvcCache.Data[subHosts[0]]
+			var allHost []string
+			for svc := range svcs {
+				subSvcs := strings.Split(svc, "/")
+				service := fmt.Sprintf("%s.%s.svc.cluster.local", subSvcs[1], subSvcs[0])
```
Comment (YonkaFang) [inline controllers/servicefence_controller.go:390]: more accurate var name

File: controllers/servicefence_controller.go [hunk 3]
```diff
@@ -376,17 +375,61 @@ func addDomainsWithHost(domains map[string]*lazyloadv1alpha1.Destinations, sf *l
...
 		// short service name without namespace, like "reviews", needs to add namespace of servicefence and "svc.cluster.local"
 		case 1:
-			fullHost = fmt.Sprintf("%s.%s.svc.cluster.local", subDomains[0], sf.Namespace)
+			fullHost = fmt.Sprintf("%s.%s.svc.cluster.local", subHosts[0], sf.Namespace)
 		// short service name with namespace, like "reviews.default", needs to add "svc.cluster.local"
 		case 2:
-			fullHost = fmt.Sprintf("%s.%s.svc.cluster.local", subDomains[0], subDomains[1])
+			fullHost = fmt.Sprintf("%s.%s.svc.cluster.local", subHosts[0], subHosts[1])
 		default:
 			log.Errorf("%s is invalid host, skip", h)
```
Comment (YonkaFang) [inline controllers/servicefence_controller.go:434]: suppport it now or later? maybe hard to distinguish from `svc.ns` format.

File: controllers/servicefence_controller.go (hunk 1 above)
Comment (cywang1905) [reply controllers/servicefence_controller.go]: diveded into two funcs
```go
	for h, strategy := range sf.Spec.Host {

		if strings.HasSuffix(h, "/*") {
			// handle namespace level host, like 'default/*'
			handleNsHost(h, domains, nsSvcCache)
		} else {
			// handle service level host, like 'a.default.svc.cluster.local' or 'a' or 'a.default'
			handleSvcHost(h, strategy, checkStatus, domains, sf)
		}

	}

File: controllers/servicefence_controller.go (hunk 2 above)
Comment (cywang1905) [reply controllers/servicefence_controller.go]: use `xxxParts` instead `subxxx` and use `fullHost` instead `service`

File: controllers/servicefence_controller.go (hunk 3 above)
Comment (cywang1905) [reply controllers/servicefence_controller.go]: will update these check logic when adapting serviceEntry in future

**Review by YonkaFang — APPROVED** [review]

**Pull request merged**